### ISSUE TITLE
Can't build fpm

### ISSUE THREAD

**Issue by minhqdao** [fortran-lang/fpm#884]
### Description

`fpm build` currently doesn't work for me on `main`. This is the error message:

```bash
fpm_strings.f90                        failed.
[  1%] Compiling...
gfortran: error: Unrecognized option: 'single-DFPM_RELEASE_VERSION=0.8.1'
gfortran: note: valid arguments to '-fcoarray=' are: lib none single
<ERROR> Compilation failed for object " src_fpm_strings.f90.o "
<ERROR>stopping due to failed compilation
STOP 1
```

It was introduced with this PR: https://github.com/fortran-lang/fpm/pull/865

I'm using:
- GNU Fortran (Homebrew GCC 12.2.0) 12.2.0
- macOS M1, Ventura 13.3.1

@perazz

### Expected Behaviour

`fpm` should build.

### Version of fpm

`main`.

### Platform and Architecture

macOS/ARM.

### Additional Information

Is the PR necessary at all? You can get the package version easily after building the model as demonstrated in https://github.com/fortran-lang/fpm/pull/876.

**Comment by perazz** [comment]
What version of fpm are you using to build fpm? your issue was fixed in #804

**Comment by minhqdao** [comment]
I see. I was still on `0.7.0` and had to update `fpm` through `conda`, which I'd forgotten. But it works now.
